### DISCUSSION TITLE
Support for "time" Unix command

### DISCUSSION THREAD

**Discussion by matteofg** [TeXworks/texworks#1044] [General]
I would like to set a tex-engine by exploiting the `time` Unix command, like

`time pdflatex sample`

How can TeXwork be configured correctly?

**Comment by stloeffler** [comment]
In the following, I'm assuming you are using Linux (or possibly macOS, though I didn't test that).
The easiest way is to add a new typesetting tool (under Edit > Preferences... > Typesetting), give it an arbitrary name (e.g., "timed pdflatex"), specify `time` as the program, and add the two arguments `pdflatex` and `$fullname`.
Note that this will use `/usr/bin/time` instead of bash's built-in `time` command and as such might have slightly different output. To get more readable output, you might want to add an additional `-p` as a first argument, or potentially specify a custom format with `-f`.
To run the typesetting through bash (and consequently use bash's built-in `time` command) instead, you could specify `/usr/bin/bash` (or equivalent) as program with the arguments `-c` and `time pdflatex $fullname`.

**Comment by matteofg** [comment]
> I'm assuming you are using Linux (or possibly macOS, though I didn't test that)

I use a macos arm... sorry I didn't specify that.

> The easiest way is to add a new typesetting tool (under Edit > Preferences... > Typesetting), give it an arbitrary name (e.g., "timed pdflatex"), specify `time` as the program, and add the two arguments `pdflatex` and `$fullname`.

With the following settings
<img width="576" alt="Screenshot 2024-02-22 at 13 51 12" src="https://github.com/TeXworks/texworks/assets/111273603/eec4b408-f1e4-4a5b-97bc-bed9d6bf6527">
I get this response:
```
time: pdflatex: No such file or directory
        0.00 real         0.00 user         0.00 sys
```

> To run the typesetting through bash (and consequently use bash's built-in `time` command) instead, you could specify `/usr/bin/bash` (or equivalent) as program with the arguments `-c` and `time pdflatex $fullname`.

With the following settings
<img width="576" alt="Screenshot 2024-02-22 at 13 52 33" src="https://github.com/TeXworks/texworks/assets/111273603/9b843e0d-bd23-4303-a906-5387c4edf410">
I get this response:
```
zsh:1: command not found: pdflatex
pdflatex latex-test.tex  0.00s user 0.00s system 26% cpu 0.003 total
```

**Comment by stloeffler** [comment]
OK, so it seems `time` is executed, but can't find `pdflatex`. You might have to specify the full path to `pdflatex` then, as macOS is particularly notorious about not setting up environmental variables in a helpful manner (especially `$PATH` needed to find programs - something TeXworks has to work around when executing the _primary_ program, but it doesn't change parameters of course).

**Comment by matteofg** [comment]
Ok... entering the full path of `pdflatex` everything works well.

Thanks!
